### PR TITLE
Add guiding principles

### PR DESCRIPTION
**Pull request by mokagio**
First brain dump of the core principles I think we should follow, and  which I've been trying to apply to our current test suite.

### CODE REVIEW

**Comment by nhisyam** [comment]
Just thinking. Why is this created as a new repo instead of being part of existing `apple-ios`?

**Comment by mokagio** [comment]
@nhisyam felt like it deserved its own one, same as for https://github.com/iflix-letsplay/swift-style-guide. Also, this gives us the chance to keep the repo open and share it with the world 💖 

We can move this in once the work is finished though, no biggie 😉 .

File: README.md [hunk 1]
```diff
@@ -1 +1,79 @@
...
+describe("BananService") {
+  context("when fetching bananas") {
+    context("if the request succeeds") {
+      it("returns bananas") { }
+    }
+    context("if the request fails") {
+      it("returns a banana error wrapping the received one") { }
+    }
+  }
+}
```
Comment (marcelmika) [inline README.md:47]: Thumbs up for the example 👍

File: README.md [hunk 2]
```diff
@@ -1 +1,79 @@
...
+```
+
+## Other Guidelines
+
+### One assertion per test
+
+Each `it` block should have only one assertion. This is in line with the _Isolated_ principle.
+
+### AAA: Arrange, Act, Assert
```
Comment (marcelmika) [inline README.md:56]: Nice

File: README.md [hunk 3]
```diff
@@ -1 +1,79 @@
...
+- **Repeatable**. If no code has been changed, multiple test runs should give the same result. Tests that fail intermittently (_flaky tests_) could be a sign of design issues in the test code itself, or worse unexpected state assumptions in the production code under test.
+- **Self-verifying**. Tests should succeed or fail in an unambiguous way. This will make the test more rigorous, and provide a clearer investigation starting point if a failure occurs.
+- **Timely**. Unit tests should be written before the production code, or at least while it is being developed. Doing this provides a faster feedback loop on the code being written, and also ensure the code itself is testable from the start.
+
+### Tests should read like sentences
+
+This is probably the most opinionated of our guidelines. For tests to be good aid to development they need to be readable. That goes beyond the code itself, having the test structure read like a sentence makes it more effective at describing behaviour.
```
Comment (marcelmika) [inline README.md]: Even though I know it's quite opinionated I think we don't need to mention it here. As soon as we all agree that this is the rule of thumb.

**Review by marcelmika — APPROVED** [review]
Great! Love it. It's easy to read, based on existing principles (FIRST, AAA), and contains examples

File: README.md (hunk 3 above)
Comment (mokagio) [reply README.md]: Good point. The doc title already states that this is an _opinionated_ set of guidelines.

I'd consider adding a paragraph about why it's opinionated in the future, but not for now. Let's keep it lean!

**Pull request merged**